--- a/runtime/src/Heap/z/zObjectAllocator.hpp
+++ b/runtime/src/Heap/z/zObjectAllocator.hpp
@@ -25,7 +25,7 @@
 
 inline uintptr_t RegionManager::AllocPinnedLocked(size_t size)
 {
-    RegionInfo* page = objectAllocators[untype(PageAge::old)]->pinnedPage.load(std::memory_order_acquire);
+    RegionInfo* page = recentPinnedRegionList.GetHeadRegion();
     return page == nullptr ? 0 : page->Alloc(size);
 }
 
