diff --git a/runtime/src/Heap/z/zHeap.cpp b/runtime/src/Heap/z/zHeap.cpp
index cedfc6231..5aa1f7993 100644
--- a/runtime/src/Heap/z/zHeap.cpp
+++ b/runtime/src/Heap/z/zHeap.cpp
@@ -98,7 +98,7 @@ private:
     FILE* file = nullptr;
 };
 
-Heap::Heap() : _young(&_page_table, &_old.forwarding_table(), &_page_allocator)
+Heap::Heap() : _young(nullptr, &_old.forwarding_table(), &_page_allocator)
 {
     _heap = this;
     RunType::InitRunTypeMap();
