diff --git a/runtime/src/Heap/Allocator/RegionManager.cpp b/runtime/src/Heap/Allocator/RegionManager.cpp
index 564d0f941..c7679857f 100644
--- a/runtime/src/Heap/Allocator/RegionManager.cpp
+++ b/runtime/src/Heap/Allocator/RegionManager.cpp
@@ -2361,7 +2361,6 @@ void RegionManager::ForwardFromRegions()
         }
         MRT_ASSERT(region->IsValidRegion(), "the head region of fromRegionList is invalid");
         ForwardRegion<G>(region);
-        CompleteRelocationRequests(region);
     }
 
     VLOG(REPORT, "forward %zu from-region units", fromRegionList.GetUnitCount());
