diff --git a/runtime/src/Heap/Allocator/RegionManager.h b/runtime/src/Heap/Allocator/RegionManager.h
index 4fe2f3fdb..da7377ac8 100644
--- a/runtime/src/Heap/Allocator/RegionManager.h
+++ b/runtime/src/Heap/Allocator/RegionManager.h
@@ -1222,7 +1222,6 @@ inline void ExecuteForwardTask(RegionManager& regionManager, RegionList& fromReg
         if (!selected.is_request()) {
             RegionInfo* region = static_cast<RegionInfo*>(selected.ordinary);
             regionManager.ForwardRegion<G>(region);
-            regionManager.CompleteRelocationRequests(region);
             continue;
         }
 
