diff --git a/runtime/src/Heap/z/zDriver.cpp b/runtime/src/Heap/z/zDriver.cpp
index db448dd05..77d25dd0e 100644
--- a/runtime/src/Heap/z/zDriver.cpp
+++ b/runtime/src/Heap/z/zDriver.cpp
@@ -82,6 +82,7 @@ void ZDriver::run_thread()
         {
             DriverLocker locker;
             const bool major = kind == GCDriverKind::MAJOR;
+            ZAbort::reset();
             if (major) ZBreakpoint::AtBeforeGC();
             abortpoint();
             const bool completed = !ZAbort::should_abort() && ExecuteDriverRequest(request);
