--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -128,7 +128,6 @@
         TracingCollector::testMarkStartState(generation, MarkStartPoint::Begin);
     }
 #endif
-    collector.flip_old_mark_start();
     ZVerify::OnColorFlip();
 #if defined(MRT_TESTABLE_INTERNALS)
     if (TracingCollector::testMarkStartState) {
@@ -148,6 +147,7 @@
         CHECK(sequence != UINT64_MAX);
         ++sequence;
     }
+    collector.flip_old_mark_start();
     PublishPhase(GC_PHASE_ENUM);
 #if defined(MRT_TESTABLE_INTERNALS)
     if (TracingCollector::testMarkStartState) {
