--- a/llvm/lib/CodeGen/CJBarrierLowering.cpp
+++ b/llvm/lib/CodeGen/CJBarrierLowering.cpp
@@ -101,7 +101,7 @@
   Value *Start = Builder.CreateLoad(I64, SPtr, Name + ".start");
   Value *End = Builder.CreateLoad(I64, EPtr, Name + ".end");
   Value *Hit = Builder.CreateAnd(Builder.CreateICmpUGE(PlaceI, Start),
-                                  Builder.CreateICmpULT(PlaceI, End), Name + ".hit");
+                                  Builder.CreateICmpULE(PlaceI, End), Name + ".hit");
   Builder.CreateCondBr(Hit, Done, Next);
 
   Builder.SetInsertPoint(Next);
