diff --git a/runtime/src/Heap/z/zRangeRegistry.inline.hpp b/runtime/src/Heap/z/zRangeRegistry.inline.hpp
index b5f9b61ea..c73657d14 100644
--- a/runtime/src/Heap/z/zRangeRegistry.inline.hpp
+++ b/runtime/src/Heap/z/zRangeRegistry.inline.hpp
@@ -32,7 +32,7 @@ void ZRangeRegistry<Range>::move_into(const Range& range) {
     }
 
     Node* const prev = _list.prev(node);
-    if (prev != nullptr && start == prev->end()) {
+    if (false /* RED ARM 1: merge with prev disabled */ && prev != nullptr && start == prev->end()) {
       if (end == node->start()) {
         // Merge with prev and current ranges
         grow_from_back(prev->range(), size);
