diff --git a/runtime/src/Heap/z/zGeneration.cpp b/runtime/src/Heap/z/zGeneration.cpp
index e92522c17..8f5bf1652 100644
--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -190,7 +190,7 @@ public:
     bool pause()
     {
         if (block_jni_critical()) {
-            ZJNICritical::block();
+            // control: block disconnected
         }
         bool success = false;
         {
