--- a/runtime/src/Heap/z/zBarrier.inline.hpp
+++ b/runtime/src/Heap/z/zBarrier.inline.hpp
@@ -425,7 +425,7 @@
 }
 
 inline zaddress ZBarrier::load_barrier_on_phantom_oop_field_preloaded(volatile zpointer* p, zpointer o) {
-  if (ZResurrection::is_blocked()) {
+  if (false) {
     return blocking_keep_alive_load_barrier_on_phantom_oop_field_preloaded(p, o);
   }
 
@@ -444,7 +444,7 @@
 }
 
 inline zaddress ZBarrier::no_keep_alive_load_barrier_on_phantom_oop_field_preloaded(volatile zpointer* p, zpointer o) {
-  if (ZResurrection::is_blocked()) {
+  if (false) {
     return blocking_load_barrier_on_phantom_oop_field_preloaded(p, o);
   }
 
