--- a/runtime/src/Heap/z/zRootsIterator.cpp
+++ b/runtime/src/Heap/z/zRootsIterator.cpp
@@ -94,7 +94,7 @@
 
 OopStorageSetIteratorStrong::OopStorageSetIteratorStrong(unsigned workers,
                                                          ZGenerationIdOptional generation)
-    : states{{{Heap::GetHeap().GetFinalizerProcessor().StrongRootStorage(), workers}}}, generation(generation)
+    : states{{{StringDedup::Instance().WeakStorage(), workers}}}, generation(generation)
 {
     (void)this->generation;
 }
@@ -104,7 +104,7 @@
     : states{{{Heap::GetHeap().GetFinalizerProcessor().WeakRootStorage(), workers},
               {Heap::GetHeap().GetExportRootStorage(), workers},
               {SyncWeakOopStorage(), workers},
-              {StringDedup::Instance().WeakStorage(), workers}}}, generation(generation) {}
+              {Heap::GetHeap().GetFinalizerProcessor().StrongRootStorage(), workers}}}, generation(generation) {}
 
 void OopStorageSetIteratorWeak::report_num_dead()
 {
