diff --git a/runtime/src/Common/BaseObject.cpp b/runtime/src/Common/BaseObject.cpp
--- a/runtime/src/Common/BaseObject.cpp
+++ b/runtime/src/Common/BaseObject.cpp
@@ -110,7 +110,6 @@ void BaseObject::ForEachRefField(const RefFieldVisitor& visitor)
 #if defined(MRT_GC_UNIT_TESTS)
                 NoteLargeArrayInitRootVisit(LargeArrayRootVisitSite::ITERATOR_SKIP, this);
 #endif
-                return;
             }
             ForEachElementInArray(this, visitor);
         } else {
diff --git a/runtime/src/ObjectModel/MArray.cpp b/runtime/src/ObjectModel/MArray.cpp
--- a/runtime/src/ObjectModel/MArray.cpp
+++ b/runtime/src/ObjectModel/MArray.cpp
@@ -210,7 +210,6 @@ void MArray::ForEachRefFieldInRange(const RefFieldVisitor& visitor, MAddress fiel
         NoteLargeArrayInitRootVisit(LargeArrayRootVisitSite::ITERATOR_SKIP,
                                     const_cast<MArray*>(this));
 #endif
-        return;
     }
     TypeInfo* componentTi = GetComponentTypeInfo();
     MIndex size = fieldEnd - fieldStart;
