diff --git a/runtime/src/Mutator/Mutator.cpp b/runtime/src/Mutator/Mutator.cpp
--- a/runtime/src/Mutator/Mutator.cpp
+++ b/runtime/src/Mutator/Mutator.cpp
@@ -449,7 +449,6 @@ void Mutator::VisitStackRoots(const RootVisitor& func, const RootVisitor& invisi
         }
     }
     StackManager::VisitStackRoots(uwContext, func, *this);
-    VisitRawObjects(visitedInvisibleRootVisitor);
     DecObserver();
     MutatorUnlock();
 }
