diff --git a/runtime/src/Mutator/Mutator.cpp b/runtime/src/Mutator/Mutator.cpp
--- a/runtime/src/Mutator/Mutator.cpp
+++ b/runtime/src/Mutator/Mutator.cpp
@@ -1031,7 +1031,6 @@ bool Mutator::DrainStackWatermark(const RootVisitor& visitor, const RootVisitor&
     if (!IsManagedContext()) {
         bool began = stackWatermark.TryBegin(epoch, owner, 0);
         if (began) {
-            VisitRawObjects(visitedInvisibleRootVisitor);
             stackWatermark.Finish(owner);
         }
         MutatorUnlock();
