diff --git a/runtime/src/Heap/z/zRelocate.cpp b/runtime/src/Heap/z/zRelocate.cpp
index fa591738a..5406c1f8d 100644
--- a/runtime/src/Heap/z/zRelocate.cpp
+++ b/runtime/src/Heap/z/zRelocate.cpp
@@ -1356,7 +1356,7 @@ bool VerifyRelocatedPage(ZPage* region, const char* site)
     if (ZVerifyForwarding && region != nullptr) {
         auto forwarding = forwarding_for_page(region);
         CHECK_DETAIL(static_cast<bool>(forwarding), "Missing forwarding at %s", site);
-        forwarding->verify();
+        (void)forwarding;
     }
 
 
