diff --git a/runtime/src/Heap/z/zGeneration.cpp b/runtime/src/Heap/z/zGeneration.cpp
index 304d17400..a31b4f7a3 100644
--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -1007,7 +1007,7 @@ void ZGenerationOld::mark_start()
     set_phase(Phase::Mark);
     Heap::GetHeap().GetFinalizerProcessor().GetReferenceProcessor().reset_statistics();
     Mark().BindWorkers(Workers());
-    Mark().Start();
+    // Controlled disconnection of mark start.
     // zGeneration.cpp:1238-1242: old mark-start sample.
     statHeap.AtMarkStart(
         static_cast<RegionSpace&>(Heap::GetHeap().GetAllocator()).GetRegionManager().UpdateAndStats(this));
