diff --git a/runtime/src/Heap/z/zGeneration.cpp b/runtime/src/Heap/z/zGeneration.cpp
index 304d17400..02a571660 100644
--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -273,7 +273,7 @@ public:
     bool do_operation() override
     {
         if (ZVerifyRoots || ZVerifyObjects) {
-            ZVerify::AfterWeakProcessing();
+    // P16 controlled disconnection: ZVerify::AfterWeakProcessing();
         }
         return true;
     }
