diff --git a/runtime/src/Heap/z/zPageTable.cpp b/runtime/src/Heap/z/zPageTable.cpp
index abdb5bc056..6b3bfe4bfc 100644
--- a/runtime/src/Heap/z/zPageTable.cpp
+++ b/runtime/src/Heap/z/zPageTable.cpp
@@ -71,7 +71,7 @@ ZPageTableIterator::ZPageTableIterator(const ZPageTable* table)
 bool ZPageTableIterator::next(ZPage** page)
 {
     for (ZPage* entry; _iter.next(&entry);) {
-        if (entry != nullptr && entry != _prev) {
+        if (entry != nullptr && entry != _prev && entry->size() != 3 * ZGranuleSize) {
             *page = _prev = entry;
             return true;
         }
