diff --git a/runtime/src/Heap/z/zAddress.cpp b/runtime/src/Heap/z/zAddress.cpp
index b1f75af6d..f54beb249 100644
--- a/runtime/src/Heap/z/zAddress.cpp
+++ b/runtime/src/Heap/z/zAddress.cpp
@@ -23,6 +23,11 @@ MRT_EXPORT uintptr_t g_cjMarkBadMask;
 MRT_EXPORT uintptr_t g_cjStoreGoodMask;
 MRT_EXPORT uintptr_t g_cjStoreBadMask;
 MRT_EXPORT size_t g_cjLoadShift;
+MRT_EXPORT uintptr_t g_cjHeapStart;
+MRT_EXPORT uintptr_t g_cjHeapEnd;
+MRT_EXPORT uintptr_t g_cjHeapRangeCount;
+MRT_EXPORT uintptr_t g_cjHeapRangeStart[kCjHeapRangeCap];
+MRT_EXPORT uintptr_t g_cjHeapRangeEnd[kCjHeapRangeCap];
 }
 namespace MapleRuntime {
 uintptr_t ZAddressHeapBase;
