diff --git a/runtime/src/UnwindStack/MachineFrame.cpp b/runtime/src/UnwindStack/MachineFrame.cpp
--- a/runtime/src/UnwindStack/MachineFrame.cpp
+++ b/runtime/src/UnwindStack/MachineFrame.cpp
@@ -39,6 +39,6 @@ bool MachineFrame::IsN2CStubFrame() const
 {
 #if defined(ENABLE_BACKWARD_PTRAUTH_CFI)
-    uintptr_t strippedIP = PtrauthStripInstPointer(reinterpret_cast<Uptr>(ip));
+    uintptr_t strippedIP = reinterpret_cast<Uptr>(ip);
     return strippedIP == reinterpret_cast<uintptr_t>(&unwindPCForN2CStub) ||
         strippedIP == reinterpret_cast<uintptr_t>(&unwindPCForExclusiveStubFull);
 #else
