--- a/runtime/src/Heap/z/zMark.cpp
+++ b/runtime/src/Heap/z/zMark.cpp
@@ -1012,7 +1012,7 @@
         const bool isYoung = region->IsYoungRegion();
 
         if (isYoung) {
-            bool wasMarked = collector->MarkEntryObject(object, entry, &ctx.Cache());
+            bool wasMarked = (collector->MarkEntryObject(object, entry, &ctx.Cache()), false);
             if (entry.mark() && wasMarked) {
                 return;
             }
