--- a/runtime/src/Heap/z/zObjectAllocator.cpp
+++ b/runtime/src/Heap/z/zObjectAllocator.cpp
@@ -225,7 +225,7 @@
 
 uintptr_t ZObjectAllocator::PerAge::alloc_large_object(size_t size, ZAllocationFlags flags, bool clearPayload)
 {
-    ZPage* page = alloc_page(ZPageType::large, AlignUp(size, ZGranuleSize), flags, clearPayload);
+    ZPage* page = alloc_page(ZPageType::large, AlignUp(size, ZGranuleSize) + ZGranuleSize, flags, clearPayload);
     return page == nullptr ? 0 : page->alloc_object(size);
 }
 
