--- a/runtime/src/Heap/z/zObjectAllocator.cpp
+++ b/runtime/src/Heap/z/zObjectAllocator.cpp
@@ -236,7 +236,7 @@
 
 uintptr_t ZObjectAllocator::PerAge::alloc_small_object(size_t size, ZAllocationFlags flags)
 {
-    return alloc_object_in_shared_page(shared_small_page_addr(), ZPageType::small, ZPageSizeSmall, size, flags);
+    return alloc_object_in_shared_page(shared_small_page_addr(), ZPageType::small, ZPageSizeSmall, size + 8, flags);
 }
 
 uintptr_t ZObjectAllocator::PerAge::alloc_object(size_t size, ZAllocationFlags flags, bool clearPayload)
