diff --git a/runtime/src/Heap/z/zPageAllocator.cpp b/runtime/src/Heap/z/zPageAllocator.cpp
index 572872050..8152f52d7 100644
--- a/runtime/src/Heap/z/zPageAllocator.cpp
+++ b/runtime/src/Heap/z/zPageAllocator.cpp
@@ -841,7 +841,7 @@ retry:
         size = request.Memory().size;
         size_t committedBytes = 0;
         ZPage* region = freeRegionManager.MaterializePageMemory(
-            request.Memory(), type, request.ExpectsPhysicalMemory(), request.ClearsPayload(), committedBytes, age);
+            request.Memory(), type == ZPageType::medium ? ZPageType::large : type, request.ExpectsPhysicalMemory(), request.ClearsPayload(), committedBytes, age);
         if (request.Memory().virtualClaimed) {
             std::lock_guard<std::mutex> lock(pageAllocatorMutex);
             const uintptr_t end = ZPage::GranuleAddress(request.Memory().index) + size;
