diff --git a/runtime/src/Heap/z/zPageAllocator.cpp b/runtime/src/Heap/z/zPageAllocator.cpp
index 572872050..46c37a488 100644
--- a/runtime/src/Heap/z/zPageAllocator.cpp
+++ b/runtime/src/Heap/z/zPageAllocator.cpp
@@ -254,7 +254,7 @@ bool FreeRegionManager::ClaimPageMemory(size_t num, PageMemory& memory, ZAllocat
 {
     std::lock_guard<std::mutex> lock(cacheMutex);
     CHECK(num != 0 && num % ZGranuleSize == 0);
-    const size_t size = num;
+    const size_t size = num; flags = ZAllocationFlags{};
     for (size_t visited = 0; visited < partitions.size(); ++visited) {
         const size_t selected = (nextPartition + visited) % partitions.size();
         Partition& partition = *partitions[selected];
