diff --git a/runtime/src/Heap/z/zAddress.inline.hpp b/runtime/src/Heap/z/zAddress.inline.hpp
index d809a0b6e..cf2f8e2f4 100644
--- a/runtime/src/Heap/z/zAddress.inline.hpp
+++ b/runtime/src/Heap/z/zAddress.inline.hpp
@@ -384,7 +384,7 @@ inline zaddress_unsafe ZOffset::address_unsafe(zoffset offset) {
 inline zaddress ZPointer::uncolor(zpointer ptr) {
   assert(ZPointer::is_load_good(ptr) || is_null_any(ptr));
   const uintptr_t raw_addr = untype(ptr);
-  return to_zaddress(raw_addr >> ZPointer::load_shift_lookup(raw_addr));
+  return to_zaddress(raw_addr & ((uintptr_t(1) << 48) - 1));
 }
 
 inline zaddress ZPointer::uncolor_store_good(zpointer ptr) {
@@ -395,7 +395,7 @@ inline zaddress ZPointer::uncolor_store_good(zpointer ptr) {
 inline zaddress_unsafe ZPointer::uncolor_unsafe(zpointer ptr) {
   assert(ZPointer::is_store_bad(ptr));
   const uintptr_t raw_addr = untype(ptr);
-  return to_zaddress_unsafe(raw_addr >> ZPointer::load_shift_lookup(raw_addr));
+  return to_zaddress_unsafe(raw_addr & ((uintptr_t(1) << 48) - 1));
 }
 
 inline bool ZPointer::is_load_bad(zpointer ptr) {
