diff --git a/runtime/src/Heap/z/zRemembered.cpp b/runtime/src/Heap/z/zRemembered.cpp
index cc51d1164..0664ef594 100644
--- a/runtime/src/Heap/z/zRemembered.cpp
+++ b/runtime/src/Heap/z/zRemembered.cpp
@@ -159,7 +159,7 @@ void ZRemembered::register_found_old(ZPage* page)
     CHECK(map.Ready());
     zoffset offset;
     CHECK(map.offset_for_address(page->GetRegionStart(), &offset));
-    _found_old.register_page(static_cast<size_t>(untype(offset)) / map.granule());
+    _found_old.register_page(static_cast<size_t>(untype(page->start())) >> ZGranuleSizeShift);
 }
 
 template<typename Function>
