diff --git a/runtime/src/Heap/z/zMark.cpp b/runtime/src/Heap/z/zMark.cpp
index 3e10d490f..f4e62215d 100644
--- a/runtime/src/Heap/z/zMark.cpp
+++ b/runtime/src/Heap/z/zMark.cpp
@@ -1192,7 +1192,7 @@ void ZMark::MarkAndFollow(MarkContext& ctx, const MarkStackEntry& entry)
     if (UNLIKELY(MarkPartialArray::IsPartialArrayEntry(entry))) {
         MarkPartialArray::FollowPartialReferences(entry, [&entry](MAddress slot) {
             auto& field = HeapSlotAt<>(slot);
-            ZBarrier::MarkBarrierOnOldOopField(nullptr, field, entry.finalizable());
+            ZBarrier::MarkBarrierOnOldOopField(nullptr, field, false);
         }, publish);
         return;
     }
