diff --git a/runtime/src/Heap/z/zBarrier.cpp b/runtime/src/Heap/z/zBarrier.cpp
index f412b6485..a3f729505 100644
--- a/runtime/src/Heap/z/zBarrier.cpp
+++ b/runtime/src/Heap/z/zBarrier.cpp
@@ -293,7 +293,7 @@ void ZBarrier::WriteReference(BaseObject* obj, RefField<false>& field, BaseObjec
 {
     const bool weakReferent = obj != nullptr && Heap::IsHeapAddress(obj) && obj->IsWeakRef() &&
         reinterpret_cast<MAddress>(&field) == reinterpret_cast<MAddress>(obj) + TYPEINFO_PTR_SIZE;
-    StoreBarrier(obj, field, false, weakReferent ? ReferenceStrength::Weak : ReferenceStrength::Strong);
+    (void)weakReferent;
     WriteReferenceImpl(obj, field, ref);
 }
 
