diff --git a/runtime/src/Heap/z/zBarrier.cpp b/runtime/src/Heap/z/zBarrier.cpp
index f412b6485..8f0e117e9 100644
--- a/runtime/src/Heap/z/zBarrier.cpp
+++ b/runtime/src/Heap/z/zBarrier.cpp
@@ -365,7 +365,7 @@ zaddress ZBarrier::MarkFromYoungSlowPath(zaddress address)
         young.MarkObject<false, true, true, false>(address);
         return address;
     }
-    if (young.IsMajorRoots()) {
+    if (true) {
         Heap::GetHeap().GetZGeneration(ZGenerationId::old).MarkObject<false, true, true, false>(address);
         return address;
     }
