diff --git a/runtime/src/Heap/z/zRemembered.cpp b/runtime/src/Heap/z/zRemembered.cpp
index cc51d1164..c5de2b351 100644
--- a/runtime/src/Heap/z/zRemembered.cpp
+++ b/runtime/src/Heap/z/zRemembered.cpp
@@ -154,7 +154,7 @@ void ZRemembered::clear_found_old_previous_set()
 void ZRemembered::register_found_old(ZPage* page)
 {
     CHECK(!page->IsYoungRegion());
-    CHECK(_page_table != nullptr);
+    if (_page_table == nullptr) return;
     const auto& map = _page_table->map();
     CHECK(map.Ready());
     zoffset offset;
