diff --git a/runtime/src/Heap/z/zRemembered.cpp b/runtime/src/Heap/z/zRemembered.cpp
index cc51d1164..dd7d16855 100644
--- a/runtime/src/Heap/z/zRemembered.cpp
+++ b/runtime/src/Heap/z/zRemembered.cpp
@@ -307,7 +307,7 @@ void ZRemembered::remap_current(ZRemsetTableIterator* iter)
 bool ZRemembered::scan_field(volatile zpointer* p) const
 {
     RefField<>& field = *reinterpret_cast<RefField<>*>(const_cast<zpointer*>(p));
-    const zaddress addr = ZBarrier::RemsetBarrierOnOopField(field);
+    const zaddress addr = zaddress::null;
     if (!is_null(addr) && Heap::is_young(untype(addr))) {
         remember(p);
         return true;
