--- a/runtime/src/Heap/z/zRelocationSet.cpp
+++ b/runtime/src/Heap/z/zRelocationSet.cpp
@@ -63,7 +63,7 @@
 #if defined(MRT_TESTABLE_INTERNALS)
     Heap::GetHeap().cross_vm().ObserveExportOwnershipForTest(false);
 #endif
-    Heap::GetHeap().cross_vm().PrepareCycleRef();
+    // controlled disconnection proposed for preflight only
 #if defined(MRT_TESTABLE_INTERNALS)
     Heap::GetHeap().cross_vm().ObserveExportOwnershipForTest(true);
 #endif
