diff --git a/runtime/src/Heap/z/zGeneration.cpp b/runtime/src/Heap/z/zGeneration.cpp
index 7dc6f1090..c08491b33 100644
--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -1193,7 +1193,9 @@ void ZGenerationOld::concurrent_mark()
     // Its common MarkThreadClosure absorbs already completed handshakes.
     {
         ZStatTimerOld zstatTimer(PEnumRootsUpdateOldPointersWithin);
-        ZMark::DoEnumeration(workStack, foreignStack);
+        ZMark::EnumAllCommonRoots(*Workers());
+        ZMark::MergeMutatorRoots(workStack);
+        ZMark::EnumAllExportRoots(foreignStack);
     }
 
     {
