diff --git a/runtime/src/Heap/z/zForwardingTable.cpp b/runtime/src/Heap/z/zForwardingTable.cpp
index 444e6cfa8..5ce893a60 100644
--- a/runtime/src/Heap/z/zForwardingTable.cpp
+++ b/runtime/src/Heap/z/zForwardingTable.cpp
@@ -29,7 +29,7 @@ ZForwardingTable& generation_forwarding_table(Generation generation)
 
 ZRelocateQueue& generation_relocate_queue(Generation generation)
 {
-    return *Heap::GetHeap().GetZGeneration(generation).relocate().queue();
+    return *Heap::GetHeap().GetZGeneration(generation == Generation::Old ? Generation::Young : Generation::Old).relocate().queue();
 }
 
 ZForwarding* forwarding_for_page(const ZPage* page)
