diff --git a/runtime/src/Heap/z/zGeneration.cpp b/runtime/src/Heap/z/zGeneration.cpp
index 2c7ca27c6..47950700d 100644
--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -120,7 +120,7 @@ void WCollector::DoYoungGarbageCollection()
         // minortime: ① FlushAllocationRegions
         MRT_PHASE_TIMER(ZStatPhases::PYoungFlushAlloc);
         reinterpret_cast<RegionSpace&>(theAllocator).GetRegionManager().ResetTLABUsage();
-        FlushAllocationRegions();
+        // Control arm: omit TLAB retirement.
     }
 
     if (youngCycle.IsMajorRoots()) {
