diff --git a/runtime/src/Heap/z/zMark.inline.hpp b/runtime/src/Heap/z/zMark.inline.hpp
index 8a66ae34a..321d4e415 100644
--- a/runtime/src/Heap/z/zMark.inline.hpp
+++ b/runtime/src/Heap/z/zMark.inline.hpp
@@ -17,9 +17,6 @@ inline void MarkDomain::MarkObject(zaddress address)
 {
     BaseObject* object = to_object(address);
     RegionInfo* page = RegionInfo::GetRegionInfoAt(raw(address));
-    if (page->IsAllocating()) {
-        return;
-    }
 
     const bool markBeforePush = gcThread;
     bool incLive = false;
