diff --git a/runtime/src/Heap/z/workerThread.cpp b/runtime/src/Heap/z/workerThread.cpp
index 682575e7f..b0e270378 100644
--- a/runtime/src/Heap/z/workerThread.cpp
+++ b/runtime/src/Heap/z/workerThread.cpp
@@ -34,7 +34,7 @@ void WorkerTaskDispatcher::coordinator_distribute_task(WorkerTask* task, uint32_
 
     // No workers are allowed to read the state variables until they have been signaled.
     _task = task;
-    _not_finished.store(num_workers, std::memory_order_relaxed);
+    _not_finished.store(1, std::memory_order_relaxed);
 
     // Dispatch 'num_workers' number of tasks.
     _start_semaphore.signal(num_workers);
