--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -96,6 +96,7 @@
         CHECK(sequence != UINT64_MAX);
         ++sequence;
     }
+    Heap::GetHeap().GetRememberedSet().FlipForMinor();
     PublishPhase(GC_PHASE_ENUM);
 #if defined(MRT_TESTABLE_INTERNALS)
     if (TracingCollector::testMarkStartState) {
@@ -110,7 +111,6 @@
 #endif
     {
         MRT_PHASE_TIMER(ZStatPhases::PYoungRemsetDrain);
-        Heap::GetHeap().GetRememberedSet().FlipForMinor();
     }
 #if defined(MRT_TESTABLE_INTERNALS)
     if (TracingCollector::testMarkStartState) {
