--- a/runtime/src/Heap/z/zGeneration.cpp
+++ b/runtime/src/Heap/z/zGeneration.cpp
@@ -130,7 +130,6 @@
         TracingCollector::testMarkStartState(generation, MarkStartPoint::Begin, markDomain);
     }
 #endif
-    collector.flip_old_mark_start();
     ZVerify::OnColorFlip();
 #if defined(MRT_TESTABLE_INTERNALS)
     if (TracingCollector::testMarkStartState) {
@@ -149,6 +148,7 @@
         CHECK(sequence != UINT64_MAX);
         ++sequence;
     }
+    collector.flip_old_mark_start();
     PublishPhase(GC_PHASE_ENUM);
 #if defined(MRT_TESTABLE_INTERNALS)
     if (TracingCollector::testMarkStartState) {
